From 13fb70fec16efca4f54788cd9c87c8ece4700d03 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Mon, 18 Oct 2021 18:36:41 +0100
Subject: [PATCH] [shell] Ignore placeholder screens

A placeholder screen is created by Qt when no real screens exist. We
don't want to create panels and containments for those, it is a whole
world of pain.

Plasma has a hook to ignore some screens which is extended with this
check.

CCBUG: 419492
---
 shell/shellcorona.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/shell/shellcorona.cpp b/shell/shellcorona.cpp
index aa7c53ba6..221c549cf 100644
--- a/shell/shellcorona.cpp
+++ b/shell/shellcorona.cpp
@@ -1126,6 +1126,9 @@ bool ShellCorona::isOutputRedundant(QScreen *screen) const
         if (screen == s) {
             continue;
         }
+        if (s->geometry().isNull()) {
+            return false;
+        }
 
         const QRect otherGeometry = s->geometry();
 
-- 
GitLab

